### PR TITLE
Remove AuthConfig worker, instead call in `PrepareDinosaurRequest` directly

### PR DESCRIPTION
**Pull request by SimonBaeumer**
## Description

**Motivation**

The AuthConfigManager is currently executed as a separate worker in its own goroutine which blocked the removal of the preparing worker in https://github.com/stackrox/acs-fleet-manager/pull/1491. 
This was because  `PrepareDinosaurRequest` blocked the [provisioning process indirectly](https://github.com/stackrox/acs-fleet-manager/pull/1493/files#diff-080490e019286b765c01a138eab3a47c8dd81e8c5c942fc8545be9d8fa8ae4e9R344-R347) which creates implicit dependencies between different workers.

In this PR the augmentation of the auth configuration is a directly done in the preparing reconciler.

**Effects**

- Remove `AuthConfigManager` is removed as worker and called directly in `PrepareDinosaurRequest` .
- Faster Central provisioning, preparing state is not dependent on another worker anymore
- The AuthConfig is still retried but as part of the preparing worker

**Breaking change**

 - The auth configuration is not reconciled on Central's in `ready` state anymore.

## Checklist (Definition of Done)

- [ ] Unit and integration tests added
- [ ] Added test description under `Test manual`
- [ ] Documentation added if necessary (i.e. changes to dev setup, test execution, ...)
- [ ] CI and all relevant tests are passing
- [ ] Add the ticket number to the PR title if available, i.e. `ROX-12345: ...`
- [ ] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.
- [ ] Add secret to app-interface Vault or Secrets Manager if necessary
- [ ] RDS changes were e2e tested [manually](../docs/development/howto-e2e-test-rds.md)
- [ ] Check AWS limits are reasonable for changes provisioning new resources

## Test manual

 - CI
 - Integration environment prober to test dynamic clients

### CODE REVIEW

**Comment by SimonBaeumer** [comment]
comment by @rukletsov:
> makes obtaining RHSSO config a blocking operation, are fine with that? And possibly removes retries.

File: internal/dinosaur/pkg/services/dinosaur.go [hunk 1]
```diff
@@ -144,6 +145,7 @@ func NewDinosaurService(connectionFactory *db.ConnectionFactory, clusterService
 		clusterPlacementStrategy: clusterPlacementStrategy,
 		amsClient:                amsClient,
 		rhSSODynamicClientsAPI:   dynamicclients.NewDynamicClientsAPI(iamConfig.RedhatSSORealm),
```
Comment (ivan-degtiarenko) [inline internal/dinosaur/pkg/services/dinosaur.go:147]: Perhaps we can also use this opportunity to move all calls to `rhSSODynamicClientsAPI` under `CentralAuthConfigManager` (or whatever the name of this structure will be) umbrella

File: internal/dinosaur/pkg/services/dinosaur.go [hunk 2]
```diff
@@ -339,13 +341,10 @@ func (k *dinosaurService) PrepareDinosaurRequest(dinosaurRequest *dbapi.CentralR
...
 	// Check IdP config is ready.
-	//
-	// TODO(alexr): Shall this go into "preparing_dinosaurs_mgr.go"? Ideally,
-	//     all CentralRequest updating logic is in one place, either in this
-	//     service or workers.
 	if dinosaurRequest.AuthConfig.ClientID == "" {
-		// We can't provision this request, skip
-		return nil
+		if err := k.centralAuthConfigManager.AddAuthConfig(dinosaurRequest); err != nil {
+			return errors.NewWithCause(errors.ErrorGeneral, err, "failed to add clientid config")
```
Comment (ivan-degtiarenko) [inline internal/dinosaur/pkg/services/dinosaur.go:346]:
```suggestion
			return errors.NewWithCause(errors.ErrorGeneral, err, "failed to add auth config")
```

File: internal/dinosaur/pkg/services/dinosaur.go [hunk 3]
```diff
@@ -339,13 +341,10 @@ func (k *dinosaurService) PrepareDinosaurRequest(dinosaurRequest *dbapi.CentralR
...
 
 	// Check IdP config is ready.
-	//
-	// TODO(alexr): Shall this go into "preparing_dinosaurs_mgr.go"? Ideally,
-	//     all CentralRequest updating logic is in one place, either in this
-	//     service or workers.
 	if dinosaurRequest.AuthConfig.ClientID == "" {
-		// We can't provision this request, skip
-		return nil
+		if err := k.centralAuthConfigManager.AddAuthConfig(dinosaurRequest); err != nil {
```
Comment (ivan-degtiarenko) [inline internal/dinosaur/pkg/services/dinosaur.go:345]: Perhaps we can replace `dinosaurRequest` -> `centralRequest`? At least within `PrepareDinosaurRequest` method

**Pull request closed without merging**